.github/workflows/lint_repo.yml

```diff
@@ -0,0 +1,18 @@
+name: Lint with Ruff
+on: push
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Install Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.11"
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install ruff
+      # Update output format to enable automatic inline annotations.
+      - name: Run Ruff
+        run: ruff check --output-format=github .
```

Comment:
@savingoyal this isn't linked to the linting choice/arrangement for metaflow, but perusing ruff more, I thought a workflow like this to be a good-to-have too!